examples/03_datetime_encoder.py

```diff
@@ -247,27 +210,25 @@
 # -------------------
 #
 # Using the |DatetimeEncoder| allows us to better understand how the date
-# impacts the NO2 concentration. To this aim, we can compute the
+# impacts the bike sharing demand. To this aim, we can compute the
 # importance of the features created by the |DatetimeEncoder|, using the
 # :func:`~sklearn.inspection.permutation_importance` function, which
 # basically shuffles a feature and sees how the model changes its prediction.
 
 ###############################################################################
 from sklearn.inspection import permutation_importance
 
-table_vec = TableVectorizer(datetime=DatetimeEncoder(add_weekday=True))
-
 # In this case, we don't use a pipeline, because we want to compute the
 # importance of the features created by the DatetimeEncoder
-X_transform = table_vec.fit_transform(X)
-feature_names = table_vec.get_feature_names_out()
+X_test_transform = pipeline[:-1].transform(X_test)
 
-model = HistGradientBoostingRegressor().fit(X_transform, y)
-result = permutation_importance(model, X_transform, y, n_repeats=10, random_state=0)
+result = permutation_importance(
+    pipeline[-1], X_test_transform, y_test, n_repeats=10, random_state=0
+)
 
 result = pd.DataFrame(
     dict(
-        feature_names=feature_names,
+        feature_names=X_test_transform.columns,
         std=result.importances_std,
         importances=result.importances_mean,
     )
```

Comment:
I would do `ascending=True` below

Reply:
and also add the std error to the barplot

result.plot.barh(
    y="importances", x="feature_names", title="Feature Importances", figsize=(12, 9), **xerr="std"**
)